nixos/modules/virtualisation/docker.nix

```diff
@@ -9,6 +9,28 @@ let
   cfg = config.virtualisation.docker;
   proxy_env = config.networking.proxy.envVars;
 
+  mkNetworkOpts = opts: concatStringsSep " "
+    ([ "--driver=${opts.driver}" ]
+    ++ optional (cfg ? subnet && cfg.subnet != null) "--subnet=${opts.subnet}"
+    ++ optional (cfg ? ip-range && cfg.ip-range != null) "--ip-range=${opts.ip-range}"
+    ++ optional (cfg ? gateway && cfg.gateway != null) "--gateway=${opts.gateway}"
+    ++ optional (cfg ? ipv6 && cfg.ipv6) "--ipv6"
+    ++ optional (cfg ? internal && cfg.internal) "--internal");
+
+
+  mkNetwork = name: opts: ''
+    id="$(${pkgs.docker}/bin/docker network ls --filter name=${name})"
+    if [[ -z "$id" ]]; then
+      ${pkgs.docker}/bin/docker network create ${mkNetworkOpts opts} ${name}
```

Comment:
should you uncreate this somewhere?

Reply:
That would involve set unions in bash. Eww. But yeah, probably.